123 cheating is easy

Fish49 opened the pull request:
More than a year after creating this branch, it has finally done what it set out to (with some hiding bugs no doubt). It hides certain game data (including from history) before sending it to clients, so tech-savvy users cant use inspect-element to see other players' cards. It also does a few other things, such as making db connections thread-safe (🤞), and setting up database versioning.

Fish49 committed:
users now cannot see opponent cards (except with move history, i plan to fix in later commit). also fixes super thick text bug and no immunity shield bug.

Fish49 committed:
Merge branch '124-convert-to-sqlite' of https://github.com/Sabacc-Organization/Sabacc-Org into 123-cheating-is-easy

Fish49 committed:
Merge branch 'main' of https://github.com/Sabacc-Organization/Sabacc-Org into 123-cheating-is-easy

Fish49 committed:
updates README.md to be more accurate and less outdated. Implements database versioning. sets database version 1 to include user preferences. moves user preferences (dark mode, theme, and card design) to the server so they wont be as finicky. TODO: move username and password away from cookies, they cause you to have to log in again quite frequently.

Fish49 committed:
makes cheating non-trivial by removing private hand data from history before serving to clients. makes db connections thread-safe by giving each thread its own connection. casts bool values in fromDb methods to ensure they are converted from the db to the python object correctly. removes global imports of old helper files from postgresql to avoid having to have a working postgresql database and to avoid having to put in your credentials for said database.

Fish49 committed:
Merge branch 'main' into 123-cheating-is-easy

Heinoushare reviewed (commented):
crazy 10x deving. How much testing has been done?

Fish49 committed:
moves TARGET_DB_VERSION to config file instead of source.  adds getConn() in some places where it used conn bare.  defines usingPsql so reading it doesn't throw an error.  fixes old gitignore problems and stops tracking client/.svelte-kit and .vscode. adds loading priority for local config files, which are not tracked by git, allowing people to have their own config.local.yml so personal data doesn't get committed by accident.

Fish49 committed:
begins removing redundant, misleading, unnecessary, or otherwise undesirable comments, along with things like console.log statements used for debugging but never cleaned up. This is unfinished work, but Im committing now so I can work on something else for a while.

Fish49 committed:
cleaning up this branch. bug fixes and small features. bug fixes:
- getClientData mutates move_history
- sabacc shifts do not happen
- move history being an empty list causes breaks on the frontend (temporary fix by not allowing None move_history on backend)
- writing kessel games to database does not work because of incompatible types (better fix possible)
- actions resulting in errors causes database desync

adds:
- client/.env.local added to .gitignore
- you can now disable verbose logging messages in the server console with the config.yaml file

todo:
- add more tooltipping to kessel games
- allow frontend to cope with empty move_history
- write .toDb methods for certain kessel attributes

Fish49 committed:
Merge branch 'main' into 123-cheating-is-easy

Fish49 committed:
cleans up merge from main to cheating-is-easy and fixes a few bugs.
bug fixes:
- bothOrAll not defined error in corellian spike
- fixes bug with sabacc pot in corellian spike
- fixes bug where corellian spike games would end one round too early (more elegant fix possible)
- fixes bug where completed and shift were 1 or 0 rather than True or False (fixed more elegantly on a different branch, merge later)
- fixed bug where app.py would ignore local.config.yml even if it existed

Heinoushare reviewed (approved):
Rigorous testing complete. A well earned PR!
